packages/core/src/utils/data.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Converts an object to an array of entries with better typing for the key
+ */
+export function objectToEntries<
+  TObject extends {
+    [Tkey in keyof TObject]: TObject[Tkey]
+  },
+>(obj: TObject) {
+  return Object.entries(obj) as [keyof TObject, TObject[keyof TObject]][]
+}
+
+/**
+ * Creates a deep clone of an object
+ */
+export const cloneDeep = <T>(obj: T): T => JSON.parse(JSON.stringify(obj))
```

Comment:
Generally not a fan of JSON stringy/parse to clone, but it might the best here.

Usually i try to manually replace using destructuring. Sadly `.structuredClone` is not available in React Native. 

We could do something with proxies, but that's also quite complex: https://ckeditor.com/blog/immutability-with-immer-clone/.